src/Extensions/Diagnostics.HealthChecks/ServiceCollectionExtensions.cs

```diff
@@ -30,12 +29,30 @@ public static IHealthChecksBuilder AddServiceFabricHealthChecks(this IServiceCol
 				});
 
 			serviceCollection.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
+		}
 
-			return serviceCollection
-				.AddHealthCheckPublisher<ServiceFabricHealthCheckPublisher>()
-				.AddHealthChecks();
+		/// <summary>
+		/// 
+		/// </summary>
+		/// <param name="serviceCollection"></param>
+		/// <returns></returns>
+		public static IHealthChecksBuilder AddServiceFabricHealthChecks<TPublisher>(this IServiceCollection serviceCollection)
+				where TPublisher : class, IHealthCheckPublisher
+		{
+			serviceCollection.ConfigureService();
+			Type publisherType = typeof(TPublisher);
+			if (publisherType == typeof(RestHealthCheckPublisher))
+			{
+				return serviceCollection.AddRestHealthCheckPublisher().AddHealthChecks();
+
+			}
+
+			return serviceCollection.AddHealthCheckPublisher<TPublisher>().AddHealthChecks();
```

Comment:
This method won't be needed at all if you would change `RestHealthCheckPublisher` provided as suggested here https://github.com/microsoft/Omex/pull/334/files#r617671328
Then both publishers could be registered by calling `AddHealthCheckPublisher`